diff --git a/utils/build-presets.ini b/utils/build-presets.ini
index ddefaff21c5..374f209e97b 100644
--- a/utils/build-presets.ini
+++ b/utils/build-presets.ini
@@ -904,6 +904,23 @@ skip-test-indexstore-db
 skip-test-sourcekit-lsp
 skip-test-swiftdocc
 
+### riscv64 ###
+[preset: mixin_buildbot_linux_riscv64]
+skip-early-swift-driver
+skip-early-swiftsyntax
+
+extra-cmake-options=
+   -DSWIFT_USE_LINKER=ld
+
+llvm-targets-to-build=host
+
+[preset: buildbot_linux_riscv64,no_test]
+mixin-preset=
+    buildbot_linux
+    mixin_buildbot_linux_riscv64
+    mixin_buildbot_linux,no_test
+### ------ ###
+
 # Linux package with out test
 [preset: buildbot_linux,no_test]
 mixin-preset=
