diff --git a/utils/build-presets.ini b/utils/build-presets.ini
index ddefaff21c5..0021e261650 100644
--- a/utils/build-presets.ini
+++ b/utils/build-presets.ini
@@ -806,6 +806,9 @@ llvm-install-components=llvm-ar;llvm-cov;llvm-profdata;IndexStore;clang;clang-re
 mixin-preset=
     mixin_linux_install_components_with_clang
 
+extra-cmake-options=
+   -DSWIFT_USE_LINKER=ld
+
 llbuild
 swiftpm
 swift-driver
