diff --git a/stdlib/cmake/modules/SwiftSource.cmake b/stdlib/cmake/modules/SwiftSource.cmake
index d5d57dfd18a..ce6503d44bc 100644
--- a/stdlib/cmake/modules/SwiftSource.cmake
+++ b/stdlib/cmake/modules/SwiftSource.cmake
@@ -570,6 +570,10 @@ function(_compile_swift_files
 
   list(APPEND swift_flags ${SWIFT_STDLIB_EXTRA_SWIFT_COMPILE_FLAGS})
 
+  if(CFLAGS_ARCH STREQUAL "riscv64")
+     list(APPEND swift_flags "-Xcc" "-mno-relax" "-Xcc" "-mabi=lp64d" "-Xcc" "-march=rv64gc")
+  endif()
+
   list(APPEND swift_flags ${SWIFT_EXPERIMENTAL_EXTRA_FLAGS})
 
   if(SWIFTFILE_OPT_FLAGS)
diff --git a/utils/build-presets.ini b/utils/build-presets.ini
index ddefaff21c5..e6c1b3849c0 100644
--- a/utils/build-presets.ini
+++ b/utils/build-presets.ini
@@ -806,6 +806,9 @@ llvm-install-components=llvm-ar;llvm-cov;llvm-profdata;IndexStore;clang;clang-re
 mixin-preset=
     mixin_linux_install_components_with_clang
 
+extra-cmake-options=
+   -DSWIFT_USE_LINKER=lld
+
 llbuild
 swiftpm
 swift-driver
