diff --git a/utils/build-presets.ini b/utils/build-presets.ini
index ddefaff21c5..da81ef1fb2d 100644
--- a/utils/build-presets.ini
+++ b/utils/build-presets.ini
@@ -1473,7 +1473,7 @@ mixin-preset=
 [preset: LLDB_Nested]
 skip-build-benchmarks
 install-destdir=%(swift_install_destdir)s
-llvm-targets-to-build=X86;ARM;AArch64;PowerPC;SystemZ;Mips
+llvm-targets-to-build=X86;ARM;AArch64;PowerPC;SystemZ;Mips;RISCV
 
 [preset: LLDB_Swift_DebugAssert]
 mixin-preset=
