diff --git a/utils/build-presets.ini b/utils/build-presets.ini
index 0c26d9e9407..1c6841d80dc 100644
--- a/utils/build-presets.ini
+++ b/utils/build-presets.ini
@@ -49,7 +49,7 @@ llbuild
 swiftpm
 swift-driver
 xctest
-libicu
+skip-build-libicu
 swiftdocc
 
 build-ninja
@@ -61,7 +61,7 @@ install-swiftpm
 install-swift-driver
 install-swiftsyntax
 install-xctest
-install-libicu
+#install-libicu
 install-prefix=/usr
 install-sourcekit-lsp
 install-swiftformat
