diff --git a/utils/build-presets.ini b/utils/build-presets.ini
index c285e0362b9..7c6248de07e 100644
--- a/utils/build-presets.ini
+++ b/utils/build-presets.ini
@@ -831,6 +831,8 @@ build-swift-static-stdlib
 build-swift-static-sdk-overlay
 build-swift-stdlib-unittest-extra
 
+skip-early-swift-driver
+skip-early-swiftsyntax
 # Executes the lit tests for the installable package that is created
 # Assumes the swift-integration-tests repo is checked out
 
