diff --git a/include/swift/Runtime/Config.h b/include/swift/Runtime/Config.h
index d6caee28d29..0119ddbbd06 100644
--- a/include/swift/Runtime/Config.h
+++ b/include/swift/Runtime/Config.h
@@ -203,10 +203,10 @@ extern uintptr_t __COMPATIBILITY_LIBRARIES_CANNOT_CHECK_THE_IS_SWIFT_BIT_DIRECTL
 // with a compiler that doesn't support swiftcall; don't define these macros
 // in that case so any incorrect usage is caught.
 #if __has_attribute(swiftcall)
-#define SWIFT_CC_swift __attribute__((swiftcall))
-#define SWIFT_CONTEXT __attribute__((swift_context))
-#define SWIFT_ERROR_RESULT __attribute__((swift_error_result))
-#define SWIFT_INDIRECT_RESULT __attribute__((swift_indirect_result))
+#define SWIFT_CC_swift 
+#define SWIFT_CONTEXT 
+#define SWIFT_ERROR_RESULT 
+#define SWIFT_INDIRECT_RESULT 
 #else
 #define SWIFT_CC_swift
 #define SWIFT_CONTEXT
