diff --git a/utils/swift_build_support/swift_build_support/products/llvm.py b/utils/swift_build_support/swift_build_support/products/llvm.py
index c0011248316..a420629e46a 100644
--- a/utils/swift_build_support/swift_build_support/products/llvm.py
+++ b/utils/swift_build_support/swift_build_support/products/llvm.py
@@ -214,7 +214,7 @@ class LLVM(cmake_product.CMakeProduct):
             if not self.is_release():
                 # On Linux build LLVM and subprojects with -gsplit-dwarf which is more
                 # space/time efficient than -g on that platform.
-                llvm_cmake_options.define('LLVM_USE_SPLIT_DWARF:BOOL', 'YES')
+                llvm_cmake_options.define('LLVM_USE_SPLIT_DWARF:BOOL', 'NO')
 
         build_targets = ['all']
 
