diff --git a/utils/build-presets.ini b/utils/build-presets.ini
index bd591736ec1..aaf0588ae01 100644
--- a/utils/build-presets.ini
+++ b/utils/build-presets.ini
@@ -847,7 +847,7 @@ llbuild
 swiftpm
 swift-driver
 xctest
-libicu
+skip-build-libicu
 swiftdocc
 
 build-ninja
@@ -859,7 +859,7 @@ install-swiftpm
 install-swift-driver
 install-swiftsyntax
 install-xctest
-install-libicu
+#install-libicu
 install-prefix=/usr
 install-sourcekit-lsp
 install-swiftformat
