diff --git a/Sources/SwiftDriver/Jobs/GenericUnixToolchain+LinkerSupport.swift b/Sources/SwiftDriver/Jobs/GenericUnixToolchain+LinkerSupport.swift
index 7238ca7c..1bcf708e 100644
--- a/Sources/SwiftDriver/Jobs/GenericUnixToolchain+LinkerSupport.swift
+++ b/Sources/SwiftDriver/Jobs/GenericUnixToolchain+LinkerSupport.swift
@@ -32,6 +32,9 @@ extension GenericUnixToolchain {
     case .x86, .x86_64, .ppc64, .ppc64le, .systemz:
       // BFD linker has issues wrt relocations against protected symbols.
       return "gold"
+    case .riscv64:
+      // For RISC-V 64, use the LLVM linker (lld).
+      return "lld"
     default:
       // Otherwise, use the default BFD linker.
       return ""
