diff --git a/utils/build-presets.ini b/utils/build-presets.ini
index 2e83a50dbbe..f3525b6ffd5 100644
--- a/utils/build-presets.ini
+++ b/utils/build-presets.ini
@@ -1006,6 +1006,32 @@ skip-early-swiftsyntax
 
 reconfigure
 
+[preset: buildbot_linux,no_test,stage0]
+build-subdir=buildbot_linux/stage0
+install-destdir=%(install_destdir)s
+release
+skip-build-benchmarks
+skip-test-cmark
+skip-early-swift-driver
+skip-early-swiftsyntax
+bootstrapping=off
+llvm-targets-to-build=X86;ARM;AArch64;RISCV
+libdispatch
+foundation
+swiftpm
+xctest
+llbuild
+swift-driver
+install-all
+
+reconfigure
+
+extra-cmake-options=
+   -DLLVM_USE_LINKER=lld
+   -DSWIFT_USE_LINKER=lld
+   -DCLANG_DEFAULT_LINKER=lld
+
+
 [preset: buildbot_linux_crosscompile_android,tools=RA,stdlib=RD,build,aarch64]
 mixin-preset=buildbot_linux_crosscompile_android,tools=RA,stdlib=RD,build
 
