diff --git a/include/swift/Runtime/Config.h b/include/swift/Runtime/Config.h
index 80326e1f334..2e4e4585d74 100644
--- a/include/swift/Runtime/Config.h
+++ b/include/swift/Runtime/Config.h
@@ -20,6 +20,8 @@
 #include "swift/Basic/Compiler.h"
 #include "swift/Runtime/CMakeConfig.h"
 
+#include <iostream>
+
 /// SWIFT_RUNTIME_WEAK_IMPORT - Marks a symbol for weak import.
 #if (__has_attribute(weak_import))
 #define SWIFT_RUNTIME_WEAK_IMPORT __attribute__((weak_import))
@@ -193,10 +195,13 @@ extern uintptr_t __COMPATIBILITY_LIBRARIES_CANNOT_CHECK_THE_IS_SWIFT_BIT_DIRECTL
 // with a compiler that doesn't support swiftcall; don't define these macros
 // in that case so any incorrect usage is caught.
 #if __has_attribute(swiftcall)
-#define SWIFT_CC_swift __attribute__((swiftcall))
-#define SWIFT_CONTEXT __attribute__((swift_context))
-#define SWIFT_ERROR_RESULT __attribute__((swift_error_result))
-#define SWIFT_INDIRECT_RESULT __attribute__((swift_indirect_result))
+std::cout << "********************************" << std::endl;
+std::cout << "**** has swiftcall support *****" << std::endl;
+std::cout << "********************************" << std::endl;
+#define SWIFT_CC_swift
+#define SWIFT_CONTEXT
+#define SWIFT_ERROR_RESULT
+#define SWIFT_INDIRECT_RESULT
 #else
 #define SWIFT_CC_swift
 #define SWIFT_CONTEXT
